api/rds.proto

```diff
@@ -797,3 +806,26 @@ message HeaderMatcher {
   // * The regex *\d{3}* does not match the value *123.456*
   google.protobuf.BoolValue regex = 3;
 }
+
+message QueryArgMatcher {
+  // Specifies the name of an arg in the requested *path*'s query string.
+  string name = 1 [(validate.rules).string.min_bytes = 1];
+
+  oneof condition {
+    option (validate.required) = true;
+
+    // If specified, indicates whether the presence or abscence of the named arg
+    // is sufficient for a match.
+    bool present = 2;
```

Comment:
can you convert this into a general NOT condition such that !query-match implies (foo!=value or foo is not present) ? 
btw, header match could also use something like this.

Reply:
Thanks, I like the idea of generalizing a not condition.  I'll simplify this diff to use the same config structure as header matching, and then we can make a negation option that applies to both.

Reply:
This is probably outside of the scope of this PR, but I think if we're going to generalize this config, we should probably allow arbitrary matchers with boolean and/or/not, and allow matching on headers, query args, path/prefix, and any other match types we add.